src/state.ts

```diff
@@ -53,19 +55,38 @@ const state = (new Proxy(overlay, {
         }
     },
 
-    /** Persist sets to storage immediately */
+    /** Persist sets to storage "immediately" */
     set(target, property, value) {
-        logger.debug("State changed!", property, value)
-        target[property] = value
-        browser.storage.local.set({ state: target } as any)
+        (async () => {
+            await locks.acquire("state")
+
+            logger.debug("State changed!", property, value)
+            target[property] = value
+            browser.storage.local.set({ state: target } as any)
+
+            // Wait for reply from each script to say that they have updated their own state
+            await Promise.all([
+                // dispatch message to all content state.ts's
+                messaging.messageAllTabs("state", "stateUpdate", [{state: target}]),
```

Comment:
If there's a tab that Tridactyl can't run on open, e.g. `about:addons`, this causes a `# Error: Could not establish connection. Receiving end does not exist.` to pop-up whenever you click in an input box.

We should probably catch those errors in `messageAllTabs`, as I don't think they'll ever be useful.